app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+                                }
+                            },100);
+
+                            clickTimer = new Timer(true);
+                            break;
+                    }
+                }
+
+                break;
+            case MotionEvent.ACTION_BUTTON_RELEASE:
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    switch (motionEvent.getActionButton()){
+                        case MotionEvent.BUTTON_PRIMARY:
+                            Timer t = new Timer(true);
+                            t.schedule(new TimerTask() {
+                                @Override
+                                public void run() {
+                                    trackpadListener.mouseButton(KEY_UP,BUTTON_LEFT);
+                                }
+                            },100);
+
+                            break;
+                        case MotionEvent.BUTTON_SECONDARY:
+                            Timer ts = new Timer(true);
+                            ts.schedule(new TimerTask() {
+                                @Override
+                                public void run() {
+                                    trackpadListener.mouseButton(KEY_UP,BUTTON_RIGHT);
+                                }
+                            },100);
+                            break;
+                    }
+                }
+                break;
+
+            case MotionEvent.ACTION_MOVE:
+                trackpadListener.mouseMove(RelativeX,RelativeY);
+
+                if(isDrag){
+                    int maxX = (maxXThreshold - minXThreshold) - autoMoveSpaceThreshold;
+                    int minX = autoMoveSpaceThreshold;
+                    int maxY = (maxYThreshold - minYThreshold) - autoMoveSpaceThreshold;
+                    int minY = autoMoveSpaceThreshold;
+                    // rawX : R : 0 ~ L : 1763  // rawY : U : 1592 ~ D:528
+                    if((x < minX || x > maxX) || (y > maxY || y < minY)){
+                        if((x < minX) || (x > maxX)){
+                            if(x < minX) leftRight = -autoMoveSense;
+                            if(x > maxX) leftRight = autoMoveSense;
+                        }else leftRight = 0;
+
+                        if((y>maxY)||(y<minY)){
+                            if(y > maxY) upDown = -autoMoveSense;
+                            if(y < minY) upDown = autoMoveSense;
+                        }else upDown = 0;
+
+                        if(autoMoveTimer == null){
+                            autoMoveTimer = new Timer(true);
+                            autoMoveTimer.schedule(new TimerTask() {
+                                @Override
+                                public void run() {
+                                    trackpadListener.mouseMove(leftRight,upDown);
+                                }
+                            }, autoMoveWait,autoMoveInterval);
```

Comment:
Can you explain what this "auto move" feature does?

Reply:
This function automatically moves the pointer in that direction when the user's finger reaches the edge of the trackpad.

This function is useful when dragging or moving a window (when the pointer cannot reach the desired location due to some space).